languages/dart/README.md

```diff
@@ -0,0 +1,170 @@
+# Dart
+
+## Primitives
+
+- [string][type-string]
+- [int][type-number]
+- [boolean][type-boolean]
+- [null][type-null]
+- [symbol][type-symbol]
+
+All primitives are [immutable][concept-immutable], i.e., they cannot be altered. It is important not to confuse a primitive itself with a variable assigned a primitive value. The variable may be reassigned a new value, but the existing value can not be changed in the ways that [objects][type-object], iterables, and [functions][type-function] can be altered.
+
+## Non-primitive types
+
+- [`object`][type-object].
+
+## Boxed types / Wrappers
+
+Except for `null` and `undefined`, all primitive values have [`object`][type-object] equivalents that wrap around the primitive values:
+
+- `String` for the [`string`][type-string] primitive
+- `Number` for the [`number`][type-number] primitive
+- `Boolean` for the [`bool`][type-boolean] primitive
+- `Symbol` for the [`symbol`][type-symbol] primitive.
+
+The wrapper's `valueOf()` method returns the primitive value.
+
+Note: Wrapped primitives (using the `new Constructor()` syntax are of `typeof` [`object`][type-object], and not their primitive type).
+
+[type-boolean]: ../../types/boolean.md
+[type-function]: ../../types/function.md
+[type-null]: ../../types/null.md
+[type-number]: ../../types/number.md
+[type-object]: ../../types/object.md
+[type-string]: ../../types/string.md
+[type-symbol]: ../../types/symbol.md
+[type-undefined]: ../../concepts/undefined.md
+
+## Global Objects (not "types")
+
+- [`Boolean`][global-object-boolean]
+- [`Date`][global-object-date]
+- [`Error`][global-object-error]
+- [`Function`][global-object-function]
+- `Iterable`
+- [`JSON`][global-object-json]
+- `Number`
+- [`Object`][global-object-object]
+- [`RegExp`][global-object-regexp]
+- `String`
+- [`Map`][global-object-map]
+- [`Set`][global-object-set]
+
+[global-object-error]: ./types/error.md
+[global-object-function]: ./types/function.md
+[global-object-json]: ./types/json.md
+[global-object-map]: ./types/map.md
+[global-object-object]: ./types/object.md
+[global-object-regexp]: ./types/regexp.md
+[global-object-set]: ./types/set.md
+
+## Expressions
+
+- [`new`][keyword-new]
+- `this`
+
+## Object-oriented concepts
```

Comment:
You could consider creating a h2 heading named `Concepts`, and then have the individual concept categories be defined as h3 headings. That way, it is easy to see the concept sections go together.